hare/eligibility/oracle_test.go

```diff
@@ -542,3 +543,100 @@ func TestOracle_Eligible2(t *testing.T) {
 	_, err := o.Eligible(context.TODO(), 100, 1, 1, types.NodeID{}, []byte{})
 	assert.Equal(t, errFoo, err)
 }
+
+func TestOracleActiveSetCacheWarming(t *testing.T) {
+	r := require.New(t)
+	types.SetLayersPerEpoch(defLayersPerEpoch)
+	o := New(&mockValueProvider{1, nil}, nil, nil, nil, defLayersPerEpoch, genActive, mockBlocksProvider{}, cfg, log.NewDefault(t.Name()))
+	mc := newMockCacher()
+	o.activesCache = mc
+	mapSize := 9
+	mp := createMapWithSize(mapSize)
+	callCount := int32(0)
+	o.getActiveSet = func(ctx context.Context, epoch types.EpochID, blocks map[types.BlockID]struct{}) (map[string]struct{}, error) {
+		// simulate slow IO
+		time.Sleep(time.Second)
```

Comment:
what sleep is for? test passes without sleep as well
do you expect something to happen during the sleep? is it possible to use synchronization in some way e.g.

```go
syncPoint := make(chan struct{})
go func() {
    syncPoint <- struct{}{} // before IO
    // getActiveSet
    syncPoint <- struct{}{} // after IO
}()
// some setup code
<-syncPoint // allow getActiveSet to execute
// more setup code
<-syncPoint
```